include/dsn/utility/metrics.h

```diff
@@ -359,4 +369,35 @@ using concurrent_counter_ptr = counter_ptr<concurrent_long_adder>;
 template <typename Adder = striped_long_adder>
 using counter_prototype = metric_prototype_with<counter<Adder>>;
 
+// A volatile counter is nearly the same as a counter, except that fetch_and_reset() is provided
+// to reset the counter atomically after its value is fetched.
+// The reason why the volatile counter is introduced is that sometimes "recent" counters are
```

Comment:
Sounds it also similar to gauge type, how to distinguish gauge, counter, and volatile counter?

Reply:
> Sounds it also similar to gauge type, how to distinguish gauge, counter, and volatile counter?

- The *gauge* is an instantaneous measurement that just provides `set()` and `get()`. The value fetched from it is what it's recently set. It's can be set at any value that arbitrarily goes up and down.
- The *counter*, as its name implies, starts out at 0 and provides `increment()` and `decrement()` to accumulate.
- The *volatile counter*, based on the general *counter*, provides "recent" semantics that can be reset atomically and efficiently after each `fetch_and_reset()` call.